ci_framework/roles/polarion/tasks/main.yml

```
@@ -0,0 +1,106 @@
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Check if variables are defined and not empty
+  ansible.builtin.assert:
+    that:
+      - cifmw_polarion_jump_repo_url is defined and cifmw_polarion_jump_repo_url != ""
+      - cifmw_polarion_test_id is defined and cifmw_polarion_test_id != ""
+      - cifmw_polarion_update_testcases is defined and cifmw_polarion_update_testcases == "false" or cifmw_polarion_update_testcases == "true"
+
+- name: Look for test result XML files in artifacts directory
+  ansible.builtin.find:
+    paths: "{{ cifmw_polarion_jump_result_dir }}"
+    patterns: "*.xml"
+    recurse: yes
+  register: xml_files
+
+- block:
+  - name: Clone Jump repository
+    ansible.builtin.git:
+      repo: "{{ cifmw_polarion_jump_repo_url }}"
+      dest: "{{ cifmw_polarion_jump_repo_dir }}"
+      version: "{{ cifmw_polarion_jump_repo_branch }}"
+      force: true
+
+  - name: Create virtualenv
+    ansible.builtin.pip:
+      name: pip
+      virtualenv: "{{ cifmw_polarion_jump_repo_dir }}/jump-venv"
+      virtualenv_command: "/usr/bin/python3 -m venv"
+      state: latest
+      extra_args: --upgrade
+    when:
+      - cifmw_polarion_jump_repo_dir is defined and cifmw_polarion_jump_repo_dir != ""
+
+  - name: Use polarion-staging
+    ansible.builtin.replace:
+      path: "{{ cifmw_polarion_jump_repo_dir }}/{{ item }}"
+      regexp: "{{ polarion_fqdn }}"
+      replace: "{{ polarion_stage_fqdn }}"
+    loop:
+      - "helpers.py"
+      - ".pylero"
+    when:
+      - polarion_fqdn is defined and polarion_fqdn != ""
+      - polarion_stage_fqdn is defined and polarion_stage_fqdn != ""
+
+  - name: Run prepare_pylero.sh
+    ansible.builtin.shell: >
+      source jump-venv/bin/activate && \
+      ./prepare_pylero.sh
+    args:
+      chdir: "{{ cifmw_polarion_jump_repo_dir }}"
+      executable: /bin/bash
+    register: pylero_output
+
+  - name: Output prepare_pylero
+    ansible.builtin.debug:
+      var: pylero_output.stdout_lines
+
+  - name: Validate XML files with xmllint
+    ansible.builtin.command: "xmllint --noout {{ item.path }}"
+    loop: "{{ xml_files.files }}"
+    register: xmllint_output
+    changed_when: false
+    failed_when: xmllint_output.rc != 0
+
+  - name: Set XML files in a fact
+    ansible.builtin.set_fact:
+      found_xml_files: "{{ xml_files.files | map(attribute='path') | join(',') }}"
+
+  - name: Update Test Runs
+    ansible.builtin.shell: >
+      source "{{ cifmw_polarion_jump_repo_dir }}/jump-venv/bin/activate" && \
+      python jump.py \
```

```suggestion
      {{ cifmw_polarion_jump_repo_dir }}/jump-venv/bin/python jump.py \
```
It should do the same, without the need to get a `shell`.